### DISCUSSION TITLE
Safari autocomplete doesn't seem to work with x-model

### DISCUSSION THREAD

**Discussion by ralphjsmit** [alpinejs/alpine#1909] [7. Bugs (archived)]
First of all, thanks for creating such an awesome tool. I dislike writing JavaScript (I'm more a PHP guy😎), but AlpineJS eased a lot of my problems, because it's the perfect solution for (relatively) small and easy JS things.

I've implemented a newsletter signup form on my website, but I noticed that there's a problem with the (iOS) safari email autocomplete. It looks like AlpineJS doesn't catch the email address. 

You can check it out live at my website, [ralphjsmit.com](https://ralphjsmit.com/), if you scroll down on any page, a popup should appear. If you do this on an iPhone or iPad (or perhaps Mac with touch bar), it fills in the field (it becomes yellow), but when I click submit, my server returns an error that it didn't receive the email. 

I already tried adding `autocomplete="off"`, but it looks like that doesn't work (anymore).

Is this indeed an error or should I implement this differently?

Many thanks!

**Comment by yabdab** [comment]
For whatever reason, this  bug is not being recognized. It is mentioned twice so far ...

https://github.com/alpinejs/alpine/discussions/1810

https://github.com/alpinejs/alpine/discussions/1862

**Reply by ralphjsmit** [reply]
Ah, I see. 

That's really strange, have you found a workaround?

**Reply by calebrunion** [reply]
> For whatever reason, this bug is not being recognized. It is mentioned twice so far ...
> 
> #1810
> 
> #1862

Actually, a total of 3 mentions:
[#1501](https://github.com/alpinejs/alpine/discussions/1501)

**Reply by ralphjsmit** [reply]
It would be great if this could be addressed..

**Reply by yabdab** [reply]
@calebporzio  is there a more appropriate place to report this?

**Comment by KevinBatdorf** [comment]
As far as I can tell, this may be intentional behavior from safari to prevent bad actors from collecting emails from autofill password manager apps/browser extensions.

Regardless, I think any PR addressing this should probably look into any security ramifications. 

I would say wait on Safari to address this, and maybe as a workaround just add a `@submit` event to your `<form>` that will check and update the missing data if needed.

**Reply by calebrunion** [reply]
> As far as I can tell, this may be intentional behavior from safari to prevent bad actors from collecting emails from autofill password manager apps/browser extensions.
> 
> Regardless, I think any PR addressing this should probably look into any security ramifications.
> 
> I would say wait on Safari to address this, and maybe as a workaround just add a `@submit` event to your `<form>` that will check and update the missing data if needed.

I hear ya; and I thought that myself at first. But this **_used_** to work in Alpine V2. The difference in versions suggests that this is unexpected behavior (a bug).

**Comment by MuzafferDede** [comment]
- #1969 should fix this
